lib/srv/uacc/uacc_linux.go

```diff
@@ -327,3 +336,27 @@ func decodeUnknownError(status int, rawUaccPathErr [uaccPathErrMaxLength]C.char)
 
 	return trace.Errorf("unknown error with code %d", status)
 }
+
+// checkSpuriousENOENT runs a C function and checks if a returned ENOENT error
+// is valid.
+func checkSpuriousENOENT(target string, f func() (C.int, error)) (C.int, error) {
+	// For reasons beyond me, updwtmp() returns ENOENT even when it's completely
+	// successful. When that happens, this function checks if f's target file was
+	// modified. If it was, assume that the ENOENT is spurious.
```

Comment:
Can we check if the entry we wanted to add is in the file, instead? Basing this off of the mtime is racey, at a minimum.